File: front/medicine/src/app/shared/models/viewModels/Dto/UserSettingsDto.ts
import { Role } from "src/app/shared/enums/Role";
import { Language } from "../../../enums/Language";
import { Sex } from "../../../enums/Sex";
import { TimeZone } from "../../../enums/Timezone";

export class UserSettingsDto {

  constructor(Id: number) {
    this.Id = Id
  }

  Id!: number;
  Language: Language = Language.en;
  Name!: string;
  TimeZone: TimeZone = TimeZone["(UTC) Coordinated Universal Time"];
  Sex: Sex = Sex.male;
  Birthday!: string;
  PhoneNumber!: string;
  Roles: Role = Role.doctor;
}

File: src/Web/Medicine.Web.UseCases/Auth/Dto/UserSettingsDto.cs
using Medicine.Entities.Enums;

namespace Medicine.Web.UseCases.Auth.Dto
{


    public record UserSettingsDto(int Id, string? Name, string Email, string PhoneNumber, SystemRole Role, Language Language, Sex Sex, DateOnly? Birthday, Entities.Enums.TimeZone TimeZone)
    {
        public UserSettingsDto() : this(0, string.Empty, string.Empty, string.Empty, SystemRole.User, Language.en, Sex.Male, default, Entities.Enums.TimeZone.CoordinatedUniversalTime)
        {
        }
    }

}
